internal/v1/handler_compose_image.go

```diff
@@ -467,6 +471,97 @@ func (h *Handlers) buildCustomRepositories(ctx echo.Context, custRepos []CustomR
 	return res, nil
 }
 
+func (h *Handlers) buildTemplateRepositories(ctx echo.Context, templateID string) ([]composer.Repository, []composer.CustomRepository, error) {
+	var customRepoIDs []string
+
+	template, err := h.server.csClient.GetTemplateByID(ctx.Request().Context(), templateID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Unable to retrieve template: %v", err)
+	}
+
+	if template == nil || template.RepositoryUuids == nil {
+		return nil, nil, fmt.Errorf("Template %v has no repositories", templateID)
+	}
+	for _, id := range *template.RepositoryUuids {
+		// Check if repo is Red Hat
+		rhRepoMap, err := h.server.csClient.GetRepositories(ctx.Request().Context(), []string{}, []string{id}, false)
+		if err != nil {
+			return nil, nil, fmt.Errorf("Unable to retrieve Red Hat repositories: %v", err)
+		}
+		// Separate Red Hat repos from custom
+		if len(rhRepoMap) > 0 {
```

Comment:
if there are Red Hat repositories in the template, should they also be considered "custom repositories"?

Reply:
I'd say no, @croissanne ?

Reply:
Yea I would think not, but I'm not entirely sure actually. @jlsherrill wdyt?

Reply:
I would think they would not be custom repositories